Adjust input value color of select2 dropdown field

nielslange opened the pull request:
### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

Closes #54583.

In #54583, @pierorocca reported that the input value of select2 dropdown fields is barely visible, when using Storefront in dark mode and the classic checkout. This issue is caused by the fact that the field value is inheriting the theme's body color. As the background color of the input values is already set to `white`, this PR sets the text color to `black` to ensure that the field values remain visible. Using a black font color was recommended by @elizaan36, based on [this comment](https://github.com/woocommerce/woocommerce/issues/54583#issuecomment-2597189571).

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions.

#### Classic theme in dark mode

1. Ensure that there's a page which is using the classic checkout.
2. Install and activate the [Storefront theme](https://wordpress.org/themes/storefront/).
3. Go to `Apperance » Customize » Backround` and set it to `#16366c`.
4. Go to `Apperance » Customize » Typography` and apply the following settings:
    - Set `Heading color` to `#ffffff`.
    - Set `Text color` to `#ffffff`.
    - Set `Link / accent color` to `#eeee22`.
    - Set `Hero heading color` to `#000000`.
    - Set `Hero text color` to `#000000`.
6.  Go to the frontend and add a product to the cart.
7. Go to the checkout page with the classic checkout.
8. Verify that the input values of the `Country / Region` dropdown are visible.
9. Repeat this test with a few other classic themes in dark mode, e.g. the [Twenty Twenty theme](https://wordpress.org/themes/twentytwenty/).

<table>
<tr>
<td valign="top" width="50%">Before:
<br><br>
<img alt="Screenshot 2025-01-17 at 15 39 48" src="https://github.com/user-attachments/assets/caa476d9-a075-410b-bb45-c1722ba4ef71" />
</td>
<td valign="top" width="50%">After:
<br><br>
<img alt="Screenshot 2025-01-17 at 15 39 01" src="https://github.com/user-attachments/assets/5d480a2d-b5ba-4440-8dd5-ea4327e55f15" />
</td>
</tr>
</table>

#### Block theme in dark mode

1. Ensure that there's a page which is using the classic checkout.
2. Install and activate the [Twenty Twenty-Five theme](https://wordpress.org/themes/twentytwentyfive/).
3. Go to `Apperance » Editor » Styles` and select the `Evening` style.
4. Go to the frontend and add a product to the cart.
5. Go to the checkout page with the classic checkout.
6. Verify that the input values of the `Country / Region` dropdown are visible.
7. Repeat this test with a few other block themes in dark mode, e.g. the [Twenty Twenty-Four theme](https://wordpress.org/themes/twentytwentyfour/).

<table>
<tr>
<td valign="top" width="50%">Before:
<br><br>
<img alt="Screenshot 2025-01-17 at 15 55 54" src="https://github.com/user-attachments/assets/aff7c9eb-ea03-43a2-897a-59397380bdc7" />
</td>
<td valign="top" width="50%">After:
<br><br>
<img alt="Screenshot 2025-01-17 at 15 56 34" src="https://github.com/user-attachments/assets/9aa72da3-8e3d-49d7-ba40-dc949dfd673e" />
</td>
</tr>
</table>

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

Set font color of select2 input fields to `black` to ensure that the values remain visible when using a dark mode theme.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

nielslange committed:
Adjust input value color of select2 dropdown field

nielslange committed:
Merge branch 'trunk' into fix/54583-select2-text-color-in-classic-checkout

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

nielslange commented:
Closing and re-opening this PR to trigger the CI pipeline.

opr reviewed (approved):
Thanks Niels, this is working for me.

nielslange committed:
Merge branch 'trunk' into fix/54583-select2-text-color-in-classic-checkout

nielslange commented:
> > As the background color of the input values is already set to white, this PR sets the text color to black to ensure that the field values remain visible. Using a black font color was recommended by @elizaan36, based on [#54583 (comment)](https://github.com/woocommerce/woocommerce/issues/54583#issuecomment-2597189571).
> 
> To be precise, the color should be set to the input field font color instead of the body text color. If the field input color is white, the color should be white. If it's black, then black. I would expect that this can vary by theme or by site builder css customizations?

Thanks for clarifying this, @pierorocca. In this case, the select input field does not have a specifically defined font color. By using `color: currentColor;`, it inherits the nearest specified color from its ancestor elements. Since the select input field itself doesn't specify a color, this inheritance continues up the chain. When using the Storefront theme with the color settings you mentioned in [[#54583](https://github.com/woocommerce/woocommerce/issues/54583#issuecomment-2597189571)](https://github.com/woocommerce/woocommerce/issues/54583#issuecomment-2597189571), this could inadvertently result in the select input color being set to white, which would conflict with its white background, rendering the text invisible. 

To ensure better control and consistency with the existing theme customizations, I changed the code to:

```css
background-color: var(--wc-form-color-background, #fff);
color: var(--wc-form-color-text, #000);
```

This approach allows themes to set the desired `--wc-form-color-background` and `--wc-form-color-text`. If defined, these CSS custom properties will be applied to the select input fields. This change aligns with the styles we're currently using for other input fields, to ensure a consistent user experience across all form elements:

https://github.com/woocommerce/woocommerce/blob/a20023cf6d9f73e968d40c71355af1761a6181bb/plugins/woocommerce/client/legacy/css/forms.scss#L72-L75

@opr Given that the select input color is not defined, can you think of an alternative solution, apart from the one I implemented?

nielslange committed:
Optimise background and font color of select fields

opr reviewed (commented):
From what I could see, the only theme that defines `--wc-form-color-background` is Twenty Twenty-One. While this PR won't introduce a regression it _probably_ won't solve it for most use cases.

In terms of alternatives, `currentColor` would be my suggestion, but as you mentioned it:

> this could inadvertently result in the select input color being set to white, which would conflict with its white background, rendering the text invisible.

So I wouldn't recommend doing that. I wonder if we could look at the `color` of the `shipping-first_name` or `email` inputs to try a "best guess" at what to do. At this point we'd have to use JS for styling which isn't really ideal.

We could also document and encourage themes to define the variables you're using above, though this won't be an immediate solution.
